proposals/text/3959-json-converter.md

```diff
@@ -0,0 +1,121 @@
+- Title: Addition of a `JsonConverter` node
+- Decision driver: @bglearning
+- Start Date: 2023-01-26
+- Proposal PR: #3959
+
+# Summary
+
+Right now we don't have a node that can take json files as input to be fed into a pipeline.
+
+Proposal: Add a `JsonConverter` node that takes in a json file, parses it, and generates `Document`s.
```

Comment:
Maybe for future development, we could also include a `JsonlConverter` that could load jsonl (Json Lines) files. The format is explained [here](https://jsonlines.org/). This would be advantageous for loading extremely large (in the range of GBs) json files since the json loading in python can be quite memory inefficient. And in the jsonl case we could process the file line-by-line which is much more memory efficient. 

Or perhaps we could even support both formats with this node using a boolean flag or autodetction of file extension? I'm not sure how often this use case would occur currently, but it would make it much easier to import large amounts of Haystack Documents from disk.

Reply:
Yes, good point! I'll add it in to the proposal.